database-commons/src/main/java/io/cdap/plugin/db/sink/ETLDBOutputFormat.java

```diff
@@ -97,8 +97,8 @@ public void close(TaskAttemptContext context) throws IOException {
           try {
             if (!emptyData) {
               getStatement().executeBatch();
-              getConnection().commit();
             }
+            getConnection().commit();
           } catch (SQLException e) {
             try {
               getConnection().rollback();
```

Comment:
We should ideally separate out the executeBatch and commit calls into two separate try-catches to ensure that we commit the previously executed batches and not rollback unnecessarily in case of exception(s) occurring at executeBatch.

Reply:
Changes done, in this case we are not committing the last batch if error occurs and pipeline will get successful with wrong no of records, there will be a warning only.

Reply:
Ohh I did not realise that this is a shared code across different sinks. Please revert to the older state of the code.

Reply:
Done